src/map.cpp

```diff
@@ -171,8 +171,15 @@ void Map::setNode(v3s16 p, MapNode & n)
 	v3s16 blockpos = getNodeBlockPos(p);
 	MapBlock *block = getBlockNoCreate(blockpos);
 	v3s16 relpos = p - blockpos*MAP_BLOCKSIZE;
+	setNode(block, relpos, n);
+}
+
+void Map::setNode(MapBlock *block, v3s16 relpos, MapNode &n)
```

Comment:
The seem to be no `Map` dependencies, hence this function would fit better into the `MapBlock` class.

Reply:
The method doesn't seem generally useful enough to be a public member of `MapBlock`. I think it should remain a private helper method.

Reply:
Also, the current implementation uses `Map::m_nodedef`.

Reply:
`NodeDefManager` is accessible almost everywhere -- in `MapBlock` that's by `m_gamedef->ndef()`.

Even though it's "just" a simple function, it is a complementary function to the node getters within `MapBlock`. Since it modifies `MapBlock` directly, having it closer to the modified data makes sense. Furthermore, `Map`/`ServerMap` generally works with absolute node positions, whereas `MapBlock` is mostly (always?) relative.

Reply:
I'd prefer a `static inline` (non-class method) function in map.cpp for this, since that's where it's used. But it doesn't really matter.

Reply:
I think I'll go with that instead.